walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,70 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // throw new UnsupportedOperationException("TODO: Implement this method.");
+    final long MINUTES_IN_A_DAY = 1440;
+    final int START_OF_DAY = TimeRange.START_OF_DAY;
+    final int END_OF_DAY = TimeRange.END_OF_DAY;
+    int endOfTimeRange = 0;
+    TimeRange previousTimeRange = TimeRange.fromStartEnd(0,0,true);
+    Collection<TimeRange> unavailableTimeRange = new ArrayList<>();
+    Collection<TimeRange> unavailableNoOverlap = new ArrayList<>();
+    Collection<TimeRange> returnCollection = new ArrayList<>();
+
+    long durationOfRequest = request.getDuration();
+    Collection<String> attendeesOfRequest = request.getAttendees();
+
+    if(attendeesOfRequest.isEmpty()){
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }else if(durationOfRequest > MINUTES_IN_A_DAY){
+      return Arrays.asList();
+    }else{
+      for(String attendee : attendeesOfRequest){
+        for(Event event : events){
+          if(event.getAttendees().contains(attendee)){
+            unavailableTimeRange.add(event.getWhen());
+          }
+        }
+      }
+      for(TimeRange timerange : unavailableTimeRange){
+        if(previousTimeRange.contains(timerange)){
+        }else if(previousTimeRange.overlaps(timerange)){
+          TimeRange newTimeRange = TimeRange.fromStartEnd(previousTimeRange.start(),timerange.end(),false);
+          unavailableNoOverlap.add(newTimeRange);
+          unavailableNoOverlap.remove(previousTimeRange);
+          previousTimeRange = newTimeRange;
+        }else{
+          unavailableNoOverlap.add(timerange);
+          previousTimeRange = timerange;
```

Comment:
Since you set `previousTimeRange = timerange` on both the `if` branch and the `else` branch, you can just move it below so it will happen for both:

```
if (...) { ... }
else (...) {... }
previousTimeRange = timerange;
```

Reply:
for the if branch, the previousTimeRange is set to the new time range which is the combined one, while for the else branch, the previousTimeRange is the one being looped therefore its different so i cant move it to the bottom.

Reply:
Ah I missed that. It's fine as is then :)